src/avram/operation.cr

```diff
@@ -1,23 +1,82 @@
 require "./validations"
+require "./callbacks"
 require "./define_attribute"
 require "./save_operation_errors"
 require "./param_key_override"
+require "./needy_initializer"
 
-class Avram::VirtualForm
-  macro inherited
-    {% raise "Avram::VirtualForm has been renamed to Avram::Operation. Please inherit from Avram::Operation." %}
-  end
-end
-
-class Avram::Operation
+abstract class Avram::Operation
```

Comment:
If we wanted an extra level of type-safety, we could make this a Generic, and ensure `run` returns `T`. 🤔  Some people may just want it to run a task and be void at the end though..